src/java.base/share/classes/jdk/internal/misc/StructureViolationExceptions.java

```diff
@@ -0,0 +1,90 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.misc;
+
+import java.lang.reflect.Constructor;
+
+/**
+ * Defines static methods to allow code in java.base to cretae or throw
+ * StructureViolationException. This class will go away when the exception
+ * moves to java.base.
+ */
+public class StructureViolationExceptions {
+    private static final Constructor<?> SVE_CTOR = structureViolationExceptionCtor();
+
+    private StructureViolationExceptions() { }
+
+    /**
+     * Creates a StructureViolationException with the given message.
+     */
+    public static RuntimeException newException(String message) {
+        if (SVE_CTOR != null) {
+            try {
+                return (RuntimeException) SVE_CTOR.newInstance(message);
+            } catch (Exception e) {
+                throw new InternalError(e);
+            }
+        } else {
+            return new RuntimeException("Structure violation exception: " + message);
+        }
+    }
+
+    /**
+     * Creates a StructureViolationException with no message.
+     */
+    public static RuntimeException newException() {
+        return newException(null);
+    }
+
+    /**
+     * Throws a StructureViolationException with the given message.
+     */
+    public static void throwException(String message) {
+        throw newException(message);
+    }
+
+    /**
+     * Throws a StructureViolationException with no message.
+     */
+    public static void throwException() {
+        throw newException(null);
+    }
+
+    /**
+     * Returns the StructureViolationException(String) constructor.
+     */
+    private static Constructor<?> structureViolationExceptionCtor() {
+        Constructor<?> ctor;
+        try {
+            Class<?> exClass = Class.forName("jdk.incubator.concurrent.StructureViolationException");
```

Comment:
Should this not be `jdk.internal.misc`?

Reply:
> Should this not be `jdk.internal.misc`? (Also, if this is the case, maybe there's a test missing that could have discovered this...)

The exception is in an incubator module, it's just that code in java.base can't statically reference it.

Reply:
Maybe it should use a `MethodHandle` fetched using `IMPL_LOOKUP` instead, in order to avoid the runtime overhead of going through `CallerSensitive` methods (`Class.forName` and `Constructor.newInstance`).

It should probably also be cached.

Reply:
> Maybe it should use a `MethodHandle` fetched using `IMPL_LOOKUP` instead, in order to avoid the runtime overhead of going through `CallerSensitive` methods (`Class.forName` and `Constructor.newInstance`).
> 
> It should probably also be cached.

It is cached. It's also a very exceptional case that only arises when the SC API (separate JEP, not here) is used incorrectly.